internal/cmd/oohelperd/http_test.go

```diff
@@ -142,3 +142,54 @@ func Test_httpMapFailure(t *testing.T) {
 		})
 	}
 }
+
+func TestParseAltSvc(t *testing.T) {
+	resp := &http.Response{
+		Header: http.Header{"Alt-Svc": []string{"h3=\":443\"; ma=3600,h2=\":443\"; ma=3600"}}}
+	authority := parseAltSvc(resp)
+	if authority != ":443" {
+		t.Fatal("parsing error alt-svc")
+	}
+
+	resp.Header["Alt-Svc"] = []string{"h2=\":443\"; ma=3600,h3=\":443\"; ma=3600"}
+	authority = parseAltSvc(resp)
+	if authority != ":443" {
+		t.Fatal("parsing error alt-svc")
+	}
+
+	resp.Header["Alt-Svc"] = []string{""}
+	authority = parseAltSvc(resp)
+	if authority != "" {
+		t.Fatal("parsing error alt-svc")
+	}
+
+	resp.Header["Alt-Svc"] = []string{"h2=\":443\"; ma=3600"}
+	authority = parseAltSvc(resp)
+	if authority != "" {
+		t.Fatal("parsing error alt-svc")
+	}
+}
+
+func TestGetHTTP3Altsvc(t *testing.T) {
+	req, _ := http.NewRequest("GET", "https://cloudflare.com", nil)
```

Comment:
Our code is already riddled with integration tests that call to outside services. Unfortunately, when you have too many alike calls, the probability that one of them would fail increases. This is burdensome.

To avoid this issue, I would instead recommend using the `http/httptest` package to create a local server that runs on localhost and that returns the desired headers. In this way, because we are using localhost, we are not going to have any flakiness caused by transient issues in GitHub Actions.

Reply:
Hm, I don't know if I understood correctly, but this test is not calling to an outside server. Instead, I fabricated the http.Request and Response to test `discoverH3Endpoint`.

Reply:
Yeah, I retract my previous comment 😇